### PR TITLE
Added AsRef implementations for Arc and Rc

### PR DESCRIPTION
**Pull request by Marwes**
The implementations are straightforward, I only hope I got the stability attributes correct(?).

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+ 7f3ae0aa26d24100379ae0f3eef29916a32f4e79

Thanks!

File: src/liballoc/arc.rs [hunk 1]
```diff
@@ -332,6 +332,15 @@ impl<T: ?Sized> Deref for Arc<T> {
     }
 }
 
+#[stable(feature = "rc_arc_as_ref", since = "1.2.0")]
```
Comment (bluss) [inline src/liballoc/arc.rs]: Shouldn't this be `rust1`?

Comment (alexcrichton) [inline src/liballoc/arc.rs]: We actually have a lint to ensure that the same feature name (e.g. `rust`) is not stabilized in multiple different versions. This allows us to know that feature names are entirely stabilized all at once in only one version (useful for future analysis)

Comment (bluss) [inline src/liballoc/arc.rs]: Aha! Yeah I thought I grepped to check but I must have misgrepped, I see that in the codebase now.

**Comment by alexcrichton** [comment]
cc https://github.com/rust-lang/rust/pull/26160, due to the unintended breakage reported in #26096 we're going to revert this for now, but please feel free to make a new PR once that's merge so we can do some more analysis of the impact!

**Pull request merged**